### ISSUE TITLE
Feature request: functions (`DATE()`) and proper `ORDER BY`

### ISSUE THREAD

**Issue by mrshu** [alanmcruickshank/sqlfluff#22]
Consider the following query:

```sql
SELECT
    col_a,
    col_b,
    date_col_a,
    date_col_b
FROM "database"."sample_table"
WHERE
    DATE(date_col_b) >= current_date
    AND length(col_a) = 4
ORDER BY date_col_a DESC
```

When saved to `a.sql` we can execute `sqlfluff` on it, which results into the following:

```
$ sqlfluff lint --rules L001,L002,L003,L004,L005,L008 -n a.sql
== [a.sql] FAIL
L:   7 | P:   1 | ???? | Found unparsable segment @ 7,1: 'WHERE\n    DATE(date_...'
L:  10 | P:  21 | ???? | Found unparsable segment @ 10,21: 'DESC\n...'
```

This is unexpected, as the query has no trouble being executed in PrestoDB (that is AWS Athena) for which it was prepared.

Could it be that the quotes (`"`) around column/database names may cause the problem? Or is it the `DATE` function?

Do let me know what I can do to help debug this even further.

Thanks!

**Comment by alanmcruickshank** [comment]
Thanks @mrshu - that's a really helpful issue report. It really helps when people lay things out clearly 😄 .

There are two things here:
1. I haven't yet implemented function calls properly, which means that the parser is getting confused by `DATE(...`.
2. I haven't implemented `ORDER BY` clauses in great detail. They detect but they don't parse properly, and it appears that there's an issue with the `DESC` keyword here.

I'm going to tag this as an enhancement at this stage because it's known functionality that we don't have yet, but I expect I'll get to this very soon.

**Comment by alanmcruickshank** [comment]
@mrshu - I've just released 0.1.4 on pypi - can you try it out on your repo and post anything that doesn't work with it. I think I've fixed the two bugs you originally posted.
